sdk/core/azure-core/inc/azure/core/http/http.hpp

```diff
@@ -197,52 +197,49 @@ namespace Azure { namespace Core { namespace Http {
     Stream, ///< Stream.
   };
 
-  // Url represent the location where a request will be performed. It can be parsed and init from
-  // a string that contains all Url parts (scheme, host, path, etc).
-  // Authority is not currently supported.
   /**
-   * @brief URL.
+   * @brief Url represents the location where a request will be performed.
+   * It can be parsed and initialized from a string that contains all URL components (scheme, host,
+   * path, etc.). Authority is not currently supported.
    */
   class Url {
-    // Let Request class to be able to set retry enabled ON
-    friend class Request;
-
   private:
     std::string m_scheme;
     std::string m_host;
     int m_port{-1};
     std::string m_encodedPath;
-    std::string m_fragment;
-    // query parameters are all decoded
-    std::map<std::string, std::string> m_queryParameters;
-    std::map<std::string, std::string> m_retryQueryParameters;
-    bool m_retryModeEnabled{false};
+    // query parameters are all encoded
+    std::map<std::string, std::string> m_encodedQueryParameters;
 
-    /*********  private static methods for all instances *******/
+    // List of default non-URL-encode chars. While URL encoding a string, do not escape any chars in
+    // this set.
+    static std::unordered_set<unsigned char> defaultNonUrlEncodeChars;
+
+  public:
+    /**
+     * @brief Decodes \p value by transforming all escaped characters to it's non-encoded value.
+     *
+     * @param value URL-encoded string.
+     * @return std::string with non-URL encoded values.
+     */
     static std::string Decode(const std::string& value);
-    static std::string EncodeHost(const std::string& host);
-    static std::string EncodePath(const std::string& path);
-    static std::string EncodeQuery(const std::string& query);
-    static std::string EncodeFragment(const std::string& fragment);
-    static std::string EncodeImpl(
-        const std::string& source,
-        const std::function<bool(int)>& should_encode);
-
-    void StartRetry()
-    {
-      m_retryModeEnabled = true;
-      m_retryQueryParameters.clear();
-    }
 
-    void StopRetry()
-    {
-      m_retryModeEnabled = false;
-      m_retryQueryParameters.clear();
-    }
+    /**
+     * @brief Encodes \p value by escaping characters to the form of %HH where HH are hex digits.
+     *
+     * @remark \p doNotEncodeSymbols arg can be used to explicitly ask this function to skip
+     * characters from encoding. For instance, using this `= -` input would prevent encoding `=`, `
+     * ` and `-`.
```

Comment:
Also, use regular quotes rather than backticks (to be consistent, elsewhere we have '?')
```suggestion
     * characters from encoding. For instance, using this `= -` input would prevent encoding '=', ' ', 
     * and '-'.
```